src/non_uniform_conformal_cubed_sphere.jl

```diff
@@ -0,0 +1,444 @@
+
+Compute the arc length (distance) between two points on the unit sphere, given their Cartesian coordinates `a₁` and
+`a₂`. The points `a₁` and `a₂` should be unit vectors (magnitude 1), representing locations on the sphere.
+"""
+function spherical_distance(a₁::AbstractVector, a₂::AbstractVector)
+    (sum(a₁.^2) ≈ 1 && sum(a₂.^2) ≈ 1) || error("a₁ and a₂ must be unit vectors")
+
+    # Compute the dot product and calculate the arccosine to find the angle.
+    cosθ = dot(a₁, a₂)
+
+    # Ensure the result is within the domain of acos due to potential floating-point errors.
+    cosθ = clamp(cosθ, -1.0, 1.0)
+
+    # Return the arc length, which is the angle between the two points.
+    return acos(cosθ)
+end
+
+
+"""
+    spherical_area_triangle(a::AbstractVector, b::AbstractVector, c::AbstractVector)
+
+Return the area of a spherical triangle on the unit sphere with vertices given by the 3-vectors `a`, `b`, and `c` whose 
+origin is the the center of the sphere. The formula was first given by Eriksson (1990).
+
+If we denote with ``A``, ``B``, and ``C`` the inner angles of the spherical triangle and with ``a``, ``b``, and ``c``
+the side of the triangle then, it has been known since Euler and Lagrange that
+``\\tan(E/2) = P / (1 + \\cos a + \\cos b + \\cos c)``, where ``E = A + B + C - π`` is the triangle's excess and
+``P = (1 - \\cos²a - \\cos²b - \\cos²c + 2 \\cos a \\cos b \\cos c)^{1/2}``.
+
+On the unit sphere, ``E`` is precisely the area of the spherical triangle. Erikkson (1990) showed that ``P`` above is
+the same as the volume defined by the vectors `a`, `b`, and `c`, that is ``P = |𝐚 \\cdot (𝐛 \\times 𝐜)|``.
+
+References
+==========
+* Eriksson, F. (1990) On the measure of solid angles, Mathematics Magazine, 63 (3), 184-187, doi:10.1080/0025570X.1990.11977515
+"""
+function spherical_area_triangle(a₁::AbstractVector, a₂::AbstractVector, a₃::AbstractVector)
+    (sum(a₁.^2) ≈ 1 && sum(a₂.^2) ≈ 1 && sum(a₃.^2) ≈ 1) || error("a₁, a₂, a₃ must be unit vectors")
+
+    tan½E = abs(dot(a₁, cross(a₂, a₃)))
+    tan½E /= 1 + dot(a₁, a₂) + dot(a₂, a₃) + dot(a₁, a₃)
+
+    return 2atan(tan½E)
+end
+
+
+"""
+    spherical_area_quadrilateral(a₁, a₂, a₃, a₄)
+
+Return the area of a spherical quadrilateral on the unit sphere whose points are given by 3-vectors, `a`, `b`, `c`, and
+`d`. The area of the quadrilateral is given as the sum of the ares of the two non-overlapping triangles. To avoid having
+to pick the triangles appropriately ensuring they are not overlapping, we compute the area of the quadrilateral as the
+half the sum of the areas of all four potential triangles formed by `a₁`, `a₂`, `a₃`, and `a₄`.
+"""
+spherical_area_quadrilateral(a::AbstractVector, b::AbstractVector, c::AbstractVector, d::AbstractVector) =
+    1/2 * (spherical_area_triangle(a, b, c) + spherical_area_triangle(a, b, d) +
+           spherical_area_triangle(a, c, d) + spherical_area_triangle(b, c, d))
```

Comment:
this is duplicated from Oceananigans
not sure what's best way forward here...

if you are proposing to move it here then code should come with tests
ideally you want to import... but that would mean cyclical dependency

where are these functions used?